src/NuGet.Clients/NuGet.Console/OutputConsole/OutputConsoleProvider.cs

```diff
@@ -65,12 +66,25 @@ public class OutputConsoleProvider : IOutputConsoleProvider
                         return new OutputConsole(vsOutputWindow, vsUIShell);
                     }
                 }, NuGetUIThreadHelper.JoinableTaskFactory);
+
+            _isServerMode = new AsyncLazy<bool>(
+                () =>
+                {
+                    return IsInServerModeAsync(CancellationToken.None);
```

Comment:
`IsInServerModeAsync` doesn't look specific to the Output Window. Should we move this to some sort of Utility class?

Reply:
Eventually. I don't want us to use these checks too often. 
My thought right now is that I don't know what kind of utility we'd benefit from. So when there's a 2nd a 3rd scenarios that needs to do this check, we'll know better where we need to put it.

Reply:
As long as you're planning to eventually factor-out things like this 😉

Reply:
Yeah, myself or anybody else, if they need this code at some point. 

I know this is just personal preference at this point(odd to try to formalize), I don't add shared code unless there are lots of utilities/more than one use case for said utiltiies.